### PR TITLE
Add BigQuery auth support for native and dbt

### PR DESCRIPTION
**Pull request by eakmanrq**
Resolves: https://github.com/TobikoData/sqlmesh/pull/648 https://github.com/TobikoData/sqlmesh/issues/270

I couldn't test all of these options but this at least provides the initial support and we can respond to bug reports.

### CODE REVIEW

**Review by tobymao — APPROVED** [review]

File: sqlmesh/dbt/target.py [hunk 1]
```diff
@@ -17,6 +20,11 @@
 from sqlmesh.utils.errors import ConfigError
 from sqlmesh.utils.pydantic import PydanticModel
 
+if sys.version_info >= (3, 9):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
```
Comment (crericha) [inline sqlmesh/dbt/target.py:26]: We use this in a few places. Is it possible to put into a util and include from there?

File: sqlmesh/dbt/target.py [hunk 2]
```diff
@@ -255,3 +265,61 @@ def to_sqlmesh(self) -> ConnectionConfig:
...
+        keyfile_json: The BigQuery keyfile as a JSON string
+        token: The BigQuery token
+        refresh_token: The BigQuery refresh token
+        client_id: The BigQuery client ID
+        client_secret: The BigQuery client secret
+        token_uri: The BigQuery token URI
+        scopes: The BigQuery scopes
+    """
+
+    schema_: str = Field(alias="dataset")
```
Comment (crericha) [inline sqlmesh/dbt/target.py:290]: schema is defined in the base class. Did you mean to overwrite it here? I believe we'll still need to alias "schema" also.

Comment (eakmanrq) [reply sqlmesh/dbt/target.py]: Yeah I didn't realize dbt supported both `schema` and `dataset` for BigQuery. Updated: https://github.com/TobikoData/sqlmesh/pull/648/commits/cbc003bec427fc35e7c0caa68bb87d7ccabcc39a

File: sqlmesh/dbt/target.py [hunk 3]
```diff
@@ -255,3 +265,71 @@ def to_sqlmesh(self) -> ConnectionConfig:
...
+        refresh_token: The BigQuery refresh token
+        client_id: The BigQuery client ID
+        client_secret: The BigQuery client secret
+        token_uri: The BigQuery token URI
+        scopes: The BigQuery scopes
+    """
+
+    type: Literal["bigquery"] = "bigquery"
+    method: t.Optional[str] = BigQueryConnectionMethod.OAUTH
+    schema_: t.Optional[str] = Field(None, alias="schema")  # type: ignore
```
Comment (crericha) [inline sqlmesh/dbt/target.py:292]: schema is inherited from the base class, so I don't think you need to define it here too.

Comment (eakmanrq) [reply sqlmesh/dbt/target.py:292]: I need it to be optional.

Comment (crericha) [reply sqlmesh/dbt/target.py:292]: schema is used within the model's fully qualified name, so it must be set. For BigQuery, we'll need to figure out how to provide schema when dataset is used instead.

Comment (eakmanrq) [reply sqlmesh/dbt/target.py:292]: This does that: https://github.com/TobikoData/sqlmesh/blob/5ec62b767ab66bec9ffb6348a74626b45c2e3322/sqlmesh/dbt/target.py#L309-L316

So `schema` is optional but it makes sure that either `schema` or `dataset` are set.

**Pull request merged**